src/algebra/group_power.lean

```diff
@@ -44,6 +44,21 @@ theorem succ_smul (a : A) (n : ℕ) : (n+1)•a = a + n•a := rfl
 @[simp] theorem pow_one (a : M) : a^1 = a := mul_one _
 @[simp] theorem add_monoid.one_smul (a : A) : 1•a = a := add_zero _
 
+@[simp] lemma pow_ite (P : Prop) [decidable P] (a : M) (b c : ℕ) :
```

Comment:
This is now a bit inconsistent.  `pow_ite` and `ite_pow` are marked simp, but `mul_ite` and `ite_mul` are not marked simp.

I think both (pow and mul/add versions) should be marked simp.  This is also what Isabelle does, as far as I've heard.  Patrick's suggestion was to *also* mark `boole_mul` as simp so that is used as a shortcut by the simplifier.  (You can just mark both as simp, the linter will complain if you've messed up the order.)

Reply:
Making `mul_ite`/`ite_mul` `simp` lemmas works pretty nicely.

On the other hand, making `add_ite`/`ite_add` `simp` lemmas is pretty unpleasant. There are a few places where we have a `finset.sum` of something like `f x + ite (P x) 1 0`. You really want `sum_add_distrib` to fire first, so then the sum of the `ite (P x) 1 0` can resolve as a cardinality. If instead `add_ite` fires first, the `f x` term gets pulled inside the `ite`, when then via `sum_ite` results in two separate summations (over P and the complement of P), and you're left with the unpleasant task of putting the two sums of `f x` back together again, humpty-dumpty style.

I *think* the only place this is nasty is in `quadratic_reciprocity.lean`, where for now I've added a `-add_ite` to a `simp` set.

My preference at this point would be to make `mul_ite` a `simp` lemma, but _not_ `add_ite`. How does that sound?

Reply:
> My preference at this point would be to make mul_ite a simp lemma, but not add_ite. How does that sound?

Seems reasonable enough to me.